packages/node_modules/@webex/webex-core/test/unit/spec/services/service-catalogs.js

```diff
@@ -73,6 +74,76 @@ describe('webex-core', () => {
           assert.isArray(serviceCatalogs.hosts);
         });
       });
+
+      describe('#map', () => {
+        let priorityLocalHosts;
+
+        beforeEach('setup hosts', () => {
+          serviceCatalogs.load(
+            ServiceCatalogs.mapRemoteCatalog({
+              catalog: SERVICE_CATALOGS[0],
+              ...fixture
+            })
+          );
+
+          const {hostCatalog} = fixture;
+
+          priorityLocalHosts = {
+            'example-service-a-name': hostCatalog['example-service-a'][0].host,
+            'example-service-b-name': hostCatalog['example-service-b'][0].host
+          };
+        });
+
+        it('should only return hosts that are active/local/priority', () => {
+          const {map} = serviceCatalogs;
+          const priorityLocalHostsKeys = Object.keys(priorityLocalHosts);
+
+          assert.isTrue(priorityLocalHostsKeys.every(
+            (key) => map[key].includes(priorityLocalHosts[key])
+          ));
+        });
+      });
+    });
+
+    describe('#clear()', () => {
+      let filter;
+      let host;
+
+      beforeEach('generate the service host class objects', () => {
+        serviceCatalogs.load(ServiceCatalogs.mapRemoteCatalog({
+          catalog: SERVICE_CATALOGS[0],
+          ...fixture
+        }));
+
+        host = serviceCatalogs.hosts[0];
+
+        filter = {
+          active: true,
+          catalog: host.catalog,
+          cluster: host.id,
+          local: true,
+          priority: true,
+          service: host.service
+        };
+      });
+
+      it('should remove all hosts when called without a filter', () => {
+        serviceCatalogs.clear();
+
+        assert.equal(serviceCatalogs.hosts.length, 0);
+      });
+
+      it('should remove only filtered hosts when called with a filter', () => {
+        serviceCatalogs.clear(filter);
```

Comment:
let's move the definition of the `filter` object to within this test since the others don't use it.

Reply:
Added an additional test that uses `filter`.